src/dwarf/units.rs

```diff
@@ -299,15 +299,64 @@ impl<'dwarf> Units<'dwarf> {
         &self,
         probe: u64,
     ) -> Result<Option<(&Function<'dwarf>, Option<gimli::DwLang>)>, gimli::Error> {
-        let units_iter = self.find_units(probe);
-        for unit in units_iter {
+        for unit in self.find_units(probe) {
             if let Some(function) = unit.find_function(probe, &self.dwarf)? {
                 return Ok(Some((function, unit.language())))
             }
         }
         Ok(None)
     }
 
+    /// Find the list of inlined functions that contain `probe`.
+    pub fn find_inlined_functions<'slf>(
+        &'slf self,
+        probe: u64,
+    ) -> Result<
+        Option<
+            impl ExactSizeIterator<
+                    Item = Result<(&'dwarf str, Option<Location<'slf>>), gimli::Error>,
+                > + 'slf,
+        >,
+        gimli::Error,
+    > {
+        for unit in self.find_units(probe) {
+            if let Some(function) = unit.find_function(probe, &self.dwarf)? {
+                let inlined_fns = function.parse_inlined_functions(unit.dw_unit(), &self.dwarf)?;
```

Comment:
we don't cache this information, right? If that's the case, wouldn't it be more efficient to pass probe address to parse_inlined_functions() and skip parsing functions/ranges that definitely don't contain `probe`? If I read the current code correctly, we first parse everything, then filter out, which is less efficient.

Reply:
We do cache the information.

Reply:
ah, that's that LazyCell stuff, right?.. I see, disregard then.

Reply:
Yep.